### DISCUSSION TITLE
Replacing ULID and KSUID with FLEXID

### DISCUSSION THREAD

**Discussion by junaway** [sensedeep/dynamodb-onetable#141] [General]
Hi,

Do you know about this new package: [flexid](https://www.npmjs.com/package/flexid)?
It seems to be both (1) a fine replacement for ULID and KSUID and (2) 10x-50x faster.

You might want to check it out.

**Comment by mobsense** [comment]
Thanks for the tip. We don't use the standard ULID and so our implementation is quite a bit faster. I like your idea of batching crypto random bytes.

**Comment by garretcharp** [comment]
Neat library but just tested and it does not work like ULID, does not sort correctly in dynamo queries.

**Reply by mobsense** [reply]
Can you give us a sample of where it fails so we can address?

**Reply by garretcharp** [reply]
Oh no the ulid works just fine. The flexid package recommended by this post is what does not sort.

**Reply by mobsense** [reply]
Good to know ;-)

**Reply by junaway** [reply]
@garretcharp Could you give me an example where the sorting fails ?

**Reply by garretcharp** [reply]
hmm now that I am testing it again I can't seem to see it happen again might've been just me miss seeing it or accidentally using a different primary key on one of the items as I was playing around

**Reply by junaway** [reply]
Thanks for checking again @garretcharp.

**Comment by junaway** [comment]
@mobsense On the contrary, when I test ULID.js from this repo I get IDs with an inverted timestamp. That is, the most-frequently-varying characters are on the left and the least-frequently-varying characters are on the right. Whereas flexid does the opposite.

```
ulid [npm]        01FGRHCBE4GW4308W1KE3939WG
ulid [onetable]   HEPWMCA510NQCKY8A1SGHY67RK
```

```
ulid [npm]         (27,659 rps)   (avg: 36μs)
ulid [onetable]   (311,675 rps)   (avg: 3μs)
```
